src/hybrid/parse/hybrid_optional_args.c

```diff
@@ -0,0 +1,177 @@
+                         &methodTarget, 1, -1,
+                         ARG_OPT_OPTIONAL,
+                         ARG_OPT_ALLOWED_VALUES, allowedCombineMethods,
+                         ARG_OPT_CALLBACK, handleCombine, ctx,
+                         ARG_OPT_POSITION, 1,
+                         ARG_OPT_END);
+
+    // GROUPBY nproperties property [property ...] [REDUCE function nargs arg [arg ...] [AS alias]] [...]
+    ArgParser_AddSubArgsV(parser, "GROUPBY", "Group results by properties with reducers",
+                         &subArgs, 1, -1,
+                         ARG_OPT_OPTIONAL,
+                         ARG_OPT_CALLBACK, handleGroupby, ctx,
+                         ARG_OPT_END);
+
+    // APPLY expression [AS alias] - apply expression to each result
+    const char *applyTarget = NULL;
+    ArgParser_AddStringV(parser, "APPLY", "Apply expression to each result",
+                         &applyTarget, 1, -1,
+                         ARG_OPT_OPTIONAL,
+                         ARG_OPT_REPEATABLE,
+                         ARG_OPT_CALLBACK, handleApply, ctx,
+                         ARG_OPT_END);
+
+    // LOAD nfields field [field ...] | LOAD * - load specific fields or all fields
+    const char *loadTarget = NULL;
+    ArgParser_AddStringV(parser, "LOAD", "Load specific fields or all fields",
+                         &loadTarget, 1, -1,
+                         ARG_OPT_OPTIONAL,
+                         ARG_OPT_CALLBACK, handleLoad, ctx,
+                         ARG_OPT_END);
+
+    // FILTER expression - filter results by expression
+    const char *filterTarget = NULL;
+    ArgParser_AddStringV(parser, "FILTER", "Filter results by expression",
+                         &filterTarget, 1, 1,
+                         ARG_OPT_OPTIONAL,
+                         ARG_OPT_CALLBACK, handleFilter, ctx,
+                         ARG_OPT_END);
+
+    // TODO: Add YIELD_SCORE_AS support for score aliasing
+
+    // Parse the arguments
+    ArgParseResult parseResult = ArgParser_Parse(parser);
+    
+    // Check for errors from callbacks
+    if (QueryError_HasError(status)) {
+        ArgParser_Free(parser);
+        return REDISMODULE_ERR; // ARG_ERROR
+    }
+    if (!parseResult.success) {
+        QueryError_SetError(status, QUERY_EPARSEARGS, ArgParser_GetErrorString(parser));
+        ArgParser_Free(parser);
+        return REDISMODULE_ERR; // ARG_ERROR
+    }
+
+    ArgParser_Free(parser);
+
+    // Handle dialect-specific validation (replicated from original)
+    if (ctx->specifiedArgs & SPECIFIED_ARG_DIALECT && ctx->reqConfig->dialectVersion < APIVERSION_RETURN_MULTI_CMP_FIRST &&
+        (*(ctx->reqFlags) & QEXEC_F_SEND_SCOREEXPLAIN)) {
+        QueryError_SetError(status, QUERY_EPARSEARGS, "EXPLAINSCORE is not supported in this dialect version");
+        return REDISMODULE_ERR;
+    }
+
+      if (!(*ctx->reqFlags & QEXEC_F_SEND_HIGHLIGHT) &&
```

Comment:
Can we give names to the booleans to better understand what activates this option? perhaps a small static function with relevant name can help with readibility